core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/TaxonomyUtils.java

```diff
@@ -58,44 +56,32 @@
  * @since 0.1.4
  */
 public class TaxonomyUtils {
-  private static final Logger log = LoggerFactory.getLogger(TaxonomyUtils.class);
 
-  private static final String ncbiEntrezUtilsURL = "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?";
+  private static final String ncbiEntrezUtilsURL = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=taxonomy&term=";
 
-  public static String checkScientificNameAtNCBI(String scientificName) {
-    try {
-      String query = ncbiEntrezUtilsURL + "db=taxonomy&term=" + URLEncoder.encode(scientificName, "UTF-8");
+  private static final Map<String, String> taxonomyCache = new HashMap<>();
+
+  public static String checkScientificNameAtNCBI(String scientificName) throws IOException {
+    if (!taxonomyCache.containsKey(scientificName)) {
+      String query = ncbiEntrezUtilsURL + URLEncoder.encode(scientificName, "UTF-8");
       final HttpClient httpclient = HttpClientBuilder.create().build();
       HttpGet httpget = new HttpGet(query);
+      HttpResponse response = httpclient.execute(httpget);
+      String out = parseEntity(response.getEntity());
+      DocumentBuilder docBuilder;
       try {
-        HttpResponse response = httpclient.execute(httpget);
-        String out = parseEntity(response.getEntity());
-        log.info(out);
-        try {
-          DocumentBuilder docBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-          Document d = docBuilder.newDocument();
-
-          TransformerFactory.newInstance().newTransformer().transform(new StreamSource(new UnicodeReader(out)), new DOMResult(d));
-
-          NodeList nl = d.getElementsByTagName("Id");
-          for (int i = 0; i < nl.getLength(); i++) {
-            Element e = (Element) nl.item(i);
-            return e.getTextContent();
-          }
-        } catch (ParserConfigurationException e) {
-          log.error("check scientific name at NCBI", e);
-        } catch (TransformerException e) {
-          log.error("check scientific name at NCBI", e);
+        docBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        Document d = docBuilder.newDocument();
+        TransformerFactory.newInstance().newTransformer().transform(new StreamSource(new UnicodeReader(out)), new DOMResult(d));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Secure this "Transformer" by either disabling external DTDs or enabling secure processing. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4435)